ibis/backends/duckdb/__init__.py

```diff
@@ -184,6 +185,57 @@ def register(
 
         return self.table(table_name)
 
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1_000_000,
+    ) -> Iterator[pa.RecordBatch]:
+        _ = self._import_pyarrow()
+        query_ast = self.compiler.to_ast_ensure_limit(
+            expr, limit, params=params
+        )
+        sql = query_ast.compile()
+
+        cursor = self.raw_sql(sql)
+
+        _reader = cursor.cursor.fetch_record_batch(chunk_size=chunk_size)
+        # Horrible hack to make sure cursor isn't garbage collected
+        # before batches are streamed out of the RecordBatchReader
+        batches = IbisRecordBatchReader(_reader, cursor)
+        return batches
+        # TODO: duckdb seems to not care about the `chunk_size` argument
+        # and returns batches in 1024 row chunks
+
+    def to_pyarrow(
+        self,
+        expr: ir.Expr,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+    ) -> pa.Table:
+        _ = self._import_pyarrow()
+        query_ast = self.compiler.to_ast_ensure_limit(
+            expr, limit, params=params
+        )
+        sql = query_ast.compile()
+
+        cursor = self.raw_sql(sql)
+        table = cursor.cursor.fetch_arrow_table()
+
+        if isinstance(expr, ir.Table):
+            return table
+        elif isinstance(expr, ir.Column):
+            # Column will be a ChunkedArray, `combine_chunks` will
+            # flatten it
+            return table.columns[0].combine_chunks()
+        elif isinstance(expr, ir.Scalar):
+            return table.columns[0].combine_chunks()[0]
```

Comment:
I think we should be able to directly index into the chunked array, could you try whether
`table.columns[0][0]` works?

Reply:
It does, thanks